spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -103,6 +103,35 @@ public void addDataUnpartitioned() {
         sql("SELECT * FROM %s ORDER BY id", tableName));
   }
 
+  @Test
+  public void deleteAndAddBackAllowed() {
```

Comment:
Nit: on test name, I think 'allowed' is fairly redundant and we can remove it?  I think all of the tests check if what they test is allowed.

Reply:
I guess there is another test called 'allowed' there, but that tests the case where the checkFlag is explicitly disabled 😅 so it has some signifiance there.

Reply:
Also optional: add a test for partitioned table.

Reply:
thank you @szehon-ho , updated per your feedback and added test for partitioned table